### PR TITLE
newest new panel collapse

### PR DESCRIPTION
**Pull request by jproberson**
~* Go under 500? Go to 0~
~* Go over 10? Go to 500+~

Now it works by:
Goes down to 500, stops, then at 300 its collapses to 0
At 300 it'll go back to 500 and back up as you keep moving over.

Also double click will collapse / open as well.

### CODE REVIEW

File: apps/web/src/app/challenge/_components/challenge-layout.tsx [hunk 1]
```diff
@@ -80,9 +80,25 @@ export function ChallengeLayout({ left, right }: ChallengeLayoutProps) {
       const newLeftWidth = ((leftWidth + dx) * 100) / divideByW;
       const newTopHeight = ((topHeight + dy) * 100) / divideByH;
 
-      window.innerWidth > 1025
-        ? (leftRef.style.width = `${newLeftWidth}%`)
-        : (leftRef.style.height = `${newTopHeight}%`);
+      if (window.innerWidth > 1025) {
```
Comment (bautistaaa) [inline apps/web/src/app/challenge/_components/challenge-layout.tsx:83]: lets avoid magic numbers and assign these to const vars

**Review by bautistaaa — CHANGES_REQUESTED** [review]

File: apps/web/src/app/challenge/_components/challenge-layout.tsx (hunk 1 above)
Comment (PickleNik) [reply apps/web/src/app/challenge/_components/challenge-layout.tsx:83]: > lets avoid magic numbers and assign these to const vars

it's the `md:` screen size from https://github.com/bautistaaa/typehero/blob/9ec2e9c813c940df5f2387a1d355713ac16f8372/apps/web/tailwind.config.ts#L17C3-L17C22

Comment (bautistaaa) [reply apps/web/src/app/challenge/_components/challenge-layout.tsx:83]: thats fine. just need to assign it to a meaningful name

**Comment by jproberson** [comment]
@bautistaaa @PickleNik 
A bit of a rework tonight.

Now it works by:
Goes down to 500, stops, then at 300 its collapses to 0
At 300 it'll go back to 500 and back up as you keep moving over.

Also double click will collapse / open as well.

File: apps/web/src/app/challenge/_components/challenge-layout.tsx (hunk 1 above)
Comment (Hacksore) [reply apps/web/src/app/challenge/_components/challenge-layout.tsx]: @PickleNik does TW give you a way to import `lg` as a value from the theme in js land?

**Comment by PickleNik** [comment]
> @bautistaaa @PickleNik A bit of a rework tonight.
> 
> Now it works by: Goes down to 500, stops, then at 300 its collapses to 0 At 300 it'll go back to 500 and back up as you keep moving over.
> 
> Also double click will collapse / open as well.

I don't like the behavior of having to drag back to 300 to un-collapse, I think un-collapsing should be instant as soon as u drag beyond 0, that's more intuitive I think

**Review by PickleNik — APPROVED** [review]
lgtm

**Pull request merged**